### PR TITLE
Added tests for parsers

### LINKED ISSUE #52: Tests for gomod parsers
After #2 is done
Tests for parsers need to be implemented

### PR DESCRIPTION
**Pull request by adelowo**
I had to move `re = regexp.MustCompile(moduleRegexp)` to a package level variable so as to prevent `MustCompile` from being called all the time.  

Fixes #52

### CODE REVIEW

File: pkg/gomod/bytes/parser_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func TestContentParser_ModuleName(t *testing.T) {
+
+	moduleName := "github.com/gomods/athens"
```
Comment (michalpristas) [inline pkg/gomod/bytes/parser_test.go]: try also with some invalid names, run data table test to check it passes with simple names, names with slashes, dashes dots ... anything you can pass to URL
and does not pass with some invalid names like '?' etc. just a few cases so we're safe.

Comment (adelowo) [reply pkg/gomod/bytes/parser_test.go]: I added those in https://github.com/adelowo/athens/blob/cc43aa74b6fb45068a3fea07105e4849c9393db5/pkg/gomod/parser_test.go#L25 ... Would just improve the test cases there

Comment (michalpristas) [reply pkg/gomod/bytes/parser_test.go]: cool, sorry missed that. i'm just skimming, i will give a thorough review in few hours

File: pkg/gomod/parser_test.go [hunk 2]
```diff
@@ -31,6 +31,10 @@ func TestParse_checkVersion(t *testing.T) {
 		{"module ", false},
 		{"module github.com/gomods/athens", false},
 		{`module "github.com/gomods/athens"`, true},
+		{`module "github.com?gomods"`, false},
```
Comment (michalpristas) [inline pkg/gomod/parser_test.go]: 👍

File: pkg/gomod/bytes/parser_test.go (hunk 1 above)
Comment (adelowo) [reply pkg/gomod/bytes/parser_test.go]: Some improvements in https://github.com/gomods/athens/pull/71/commits/1502473729053f0980556ff9de1cf5298b8414c8

File: pkg/gomod/file/parser_test.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+func TestFileParser_FileNotFound(t *testing.T) {
+	parser := NewFileParser("../../../go.mod")
+
+	_, err := parser.ModuleName()
+	if !os.IsNotExist(err) {
+		t.Fatalf("file does not exist but got %v instead", err)
+	}
+}
```
Comment (arschles) [inline pkg/gomod/file/parser_test.go]: nit/note for next time: let's write tests with [assert](https://godoc.org/github.com/stretchr/testify/assert)/[require](https://godoc.org/github.com/stretchr/testify/require) helpers and, as packages start to get bigger, use [suite](https://godoc.org/github.com/stretchr/testify/suite).

Overall, these packages shorten test code and make it more readable. Not blocking for now

File: pkg/gomod/parser_test.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package parser
+
+import (
+	"errors"
+	"strings"
+	"testing"
+)
+
+func TestParse_ScannerError(t *testing.T) {
+
```
Comment (arschles) [inline pkg/gomod/parser_test.go]: can you remove this newline?

**Comment by arschles** [comment]
@adelowo #60 caused a few conflicts on this PR - ping me and I'll help out fixing them

**Comment by adelowo** [comment]
Thanks for helping out... I should be able to resolve them

**Comment by adelowo** [comment]
By the way #72 was merged... This is no longer needed anyways

File: pkg/gomod/file/parser_test.go (hunk 3 above)
Comment (adelowo) [reply pkg/gomod/file/parser_test.go]: Got that

**Comment by arschles** [comment]
@adelowo ok, thanks for letting me know 😄

**Pull request closed without merging**